components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/verify/DigitalAssetLinksHandler.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.customtabs.verify
+
+import android.webkit.URLUtil
+import androidx.annotation.VisibleForTesting
+import androidx.core.net.toUri
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Digital Asset Links APIs allows any caller to check pre declared
+ * relationships between two assets which can be either web domains
+ * or native applications. This requests checks for a specific
+ * relationship declared by a web site with an Android application
+ */
+internal class DigitalAssetLinksHandler(
+    private val httpClient: Client
+) {
+
+    fun checkDigitalAssetLinkRelationship(
+        webDomain: String,
+        packageName: String,
+        fingerprint: String?,
+        relationship: String
+    ): Boolean {
+        val requestUrl = getUrlForCheckingRelationship(webDomain, packageName, fingerprint, relationship)
+        if (!URLUtil.isValidUrl(requestUrl)) return false
+
+        val response = httpClient.fetch(Request(requestUrl))
```

Comment:
We need to catch the IOException here! This is an easy mistake that I made too since Kotlin does not enforce catching exceptions. I almost wonder whether we should change the return type to something like [Try](https://arrow-kt.io/docs/arrow/core/try/) to enforce thinking about this case.